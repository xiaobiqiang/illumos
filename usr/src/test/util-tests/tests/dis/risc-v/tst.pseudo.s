/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of various supported pseudo instructions. We only
 * support disassembling a subset of the common pseudo instructions that map
 * directly to a single asm instruction. Several of the pseudo-instructions
 * transform into more than one instruction so we don't support them.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	nop
	ret
	fence
	rdinstret	t0
	rdcycle		t1
	rdtime		t2
	csrr		t1, ustatus
	csrw		ustatus, t1
	csrs		ustatus, t2
	csrc		ustatus, t3
	csrwi		uie, 0x4
	csrsi		uie, 0x5
	csrci		uie, 0x6
	frcsr		s0
	fscsr		s0, s1
	fscsr		s1
	frrm		a0
	fsrm		a0, a1
	fsrm		a1
	fsrmi		t0, 0x4
	fsrmi		0x5	
	frflags		a0
	fsflags		a0, a1
	fsflags		a1
	fsflagsi	t0, 0x4
	fsflagsi	0x5	
.size libdis_test, [.-libdis_test]
