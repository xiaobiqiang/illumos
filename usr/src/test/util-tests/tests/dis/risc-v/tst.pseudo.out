    libdis_test:      13 00 00 00        nop
    libdis_test+0x4:  67 80 00 00        ret
    libdis_test+0x8:  0f 00 f0 0f        fence
    libdis_test+0xc:  f3 22 20 c0        rdinstret t0
    libdis_test+0x10: 73 23 00 c0        rdcycle t1
    libdis_test+0x14: f3 23 10 c0        rdtime t2
    libdis_test+0x18: 73 23 00 00        csrr t1,ustatus
    libdis_test+0x1c: 73 10 03 00        csrw ustatus,t1
    libdis_test+0x20: 73 a0 03 00        csrs ustatus,t2
    libdis_test+0x24: 73 30 0e 00        csrc ustatus,t3
    libdis_test+0x28: 73 50 42 00        csrwi uie,0x4
    libdis_test+0x2c: 73 e0 42 00        csrsi uie,0x5
    libdis_test+0x30: 73 70 43 00        csrci uie,0x6
    libdis_test+0x34: 73 24 30 00        frcsr s0
    libdis_test+0x38: 73 94 34 00        fscsr s0,s1
    libdis_test+0x3c: 73 90 34 00        fscsr s1
    libdis_test+0x40: 73 25 20 00        frrm a0
    libdis_test+0x44: 73 95 25 00        fsrm a0,a1
    libdis_test+0x48: 73 90 25 00        fsrm a1
    libdis_test+0x4c: f3 52 22 00        fsrmi t0,0x4
    libdis_test+0x50: 73 d0 22 00        fsrmi 0x5
    libdis_test+0x54: 73 25 10 00        frflags a0
    libdis_test+0x58: 73 95 15 00        fsflags a0,a1
    libdis_test+0x5c: 73 90 15 00        fsflags a1
    libdis_test+0x60: f3 52 12 00        fsflagsi t0,0x4
    libdis_test+0x64: 73 d0 12 00        fsflagsi 0x5
